internal/config/config.go

```diff
@@ -289,6 +289,8 @@ type Config struct {
 
 	// We currently only support large/small as values here.
 	Models map[SelectedModelType]SelectedModel `json:"models,omitempty" jsonschema:"description=Model configurations for different model types,example={\"large\":{\"model\":\"gpt-4o\",\"provider\":\"openai\"}}"`
+	// Recently used models stored in the data directory config.
+	RecentModels map[SelectedModelType][]SelectedModel `json:"recent_models,omitempty" jsonschema:"description=Recently used models sorted by most recent first"`
```

Comment:
I wonder if we should really save recent model on the config.

I suggest to fetch recent models from the database. This query worked great for me:

```sql
SELECT DISTINCT provider, model
FROM messages
WHERE provider <> '' AND model <> ''
ORDER BY created_at DESC
LIMIT 5
```

Let me know what you think. One disadvantage is that this will get recent models per project only. Won't work globally. But maybe it's a good trade-off to avoid having to use the config for this.

If we really want to save globally, we can also consider a separate `recent_models.json` file, as the config is in theory something the user will edit and should have relevant fields only.

Reply:
I'd personally _like_ the recent models available across all sessions, including new ones, but sourcing the info from the database once created would definitely be simpler.

Note that my PR _shouldn't_ store the data in the main config the user might read/edit at `~/.config/crush/crush.json` or the Windows equivalent. Instead, it should be in `~/.local/share/crush/crush.json`, which the README says is for _ephemeral data, such as application state_. I thought that would be the best place for it, alongside the currently-selected model and its reasoning settings and some other misc stuff the user should only modify through the TUI.

Reply:
From a UX standpoint, I agree that favorite models this should be aggregated globally versus per-project. No opinions on execution, other than that it should be, indeed, treated as ephemeral data.